src/test/regress/expected/validate_constraint.out

```diff
@@ -0,0 +1,150 @@
+                               json_array_elements_text((run_command_on_workers($$
+    SELECT
+      COALESCE(json_agg(row_to_json(d)), '[]'::json)
+    FROM
+      (
+        SELECT conname as name,
+        convalidated as validated
+        FROM pg_catalog.pg_constraint AS con
+        JOIN pg_catalog.pg_namespace AS ns
+        ON ns.oid = con.connamespace
+        WHERE nspname = 'validate_constraint'
+        AND contype = 'c'
+        ORDER BY 1,2
+      )
+      d $$)).RESULT :: json) :: json)).*
+    ORDER BY 1,2;
+CREATE VIEW constraint_validations AS
+  SELECT conname as "Constraint",
+         convalidated as "Validated?"
+  FROM pg_catalog.pg_constraint AS con
+         JOIN pg_catalog.pg_namespace AS ns ON ns.oid = con.connamespace
+  WHERE nspname = 'validate_constraint'
+    AND contype = 'c';
+CREATE TABLE referenced_table (id int UNIQUE, test_column int);
+SELECT create_reference_table('referenced_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+CREATE TABLE referencing_table (id int, ref_id int);
+SELECT create_distributed_table('referencing_table', 'ref_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE constrained_table (id int, constrained_column int);
+SELECT create_distributed_table('constrained_table', 'constrained_column');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- The two constraint types that are allowed to be NOT VALID
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT check_constraint CHECK (constrained_column > 100) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT check_constraint;
+ROLLBACK;
+BEGIN;
+ALTER TABLE referencing_table
+  ADD CONSTRAINT fk_constraint FOREIGN KEY (ref_id) REFERENCES referenced_table (id) NOT VALID;
+ALTER TABLE referencing_table
+  VALIDATE CONSTRAINT fk_constraint;
+ROLLBACK;
+-- It is possible that some other table_constraint commands will support NOT VALID option in the future
+-- These should fail as Postgres does not yet allow these constraints to be NOT VALID
+-- If one of these queries are not failing on a future Postgres version, we will need to test corresponding VALIDATE CONSTRAINT queries
+ALTER TABLE constrained_table
+  ADD CONSTRAINT unique_constraint UNIQUE (constrained_column) NOT VALID;
+ERROR:  UNIQUE constraints cannot be marked NOT VALID
+ALTER TABLE constrained_table
+  ADD CONSTRAINT pk_constraint PRIMARY KEY (id) NOT VALID;
+ERROR:  PRIMARY KEY constraints cannot be marked NOT VALID
+ALTER TABLE constrained_table
+  ADD CONSTRAINT exclude_constraint EXCLUDE USING gist (id WITH =, constrained_column WITH <>) NOT VALID;
+ERROR:  EXCLUDE constraints cannot be marked NOT VALID
+INSERT INTO constrained_table
+SELECT x, x
+from generate_series(1, 1000) as f (x);
+-- a constraint that can be validated
+ALTER TABLE constrained_table
+  ADD CONSTRAINT validatable_constraint CHECK (constrained_column < 10000) NOT VALID;
+ALTER TABLE constrained_table
```

Comment:
I think, before executing `VALIDATE CONSTRAINT`, we should query the views `constraint_validations ` and `constraint_validations_in_workers ` to ensure that we're successfully added invalid constraints.